agent/agent.go

```diff
@@ -644,33 +640,35 @@ func (a *agent) reportLifecycleUntilClose() {
 			if lastIndex == lastReportedIndex {
 				break
 			}
+			l, err := agentsdk.ProtoFromLifecycle(report)
+			if err != nil {
+				a.logger.Critical(ctx, "failed to convert lifecycle state", slog.F("report", report))
+				// Skip this report; there is no point retrying.  Maybe we can successfully convert the next one?
+				lastReportedIndex++
+				continue
+			}
+			logger := a.logger.With(slog.F("payload", l))
```

Comment:
Nit: The `payload` should be `payload := &proto.UpdateLifecycleRequest{Lifecycle: l}`, as that's what's being sent.